Generalize descriptor processing to remove hardcoding in process-css-properties.py

weinig opened the pull request:
#### 9d2d5fe065c1c94361b38a94dffc95133e4a35ba
<pre>
Generalize descriptor processing to remove hardcoding in process-css-properties.py
<a href="https://bugs.webkit.org/show_bug.cgi?id=249730">https://bugs.webkit.org/show_bug.cgi?id=249730</a>
rdar://103605497

Reviewed by Antti Koivisto.

Rather than adding explicit support for each new rule type (e.g.
@font-face, @counter-styles, etc.) to process-css-properties.py,
this generalizes all descriptors and utilizes the fact that the
rule type is present in the json already. While it is not common
to add new rule types, there was also no reason to hard code them.

* Source/WebCore/css/process-css-properties.py:

Canonical link: <a href="https://commits.webkit.org/258345@main">https://commits.webkit.org/258345@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca44bf35e8c89a53140acf95aad6ae52ef4987f4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101585 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10740 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34646 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110903 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171118 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105565 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11696 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1643 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93978 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108677 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107367 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8916 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92161 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/35438 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90808 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23579 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78437 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4338 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25079 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4407 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1534 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10493 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44567 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5724 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6167 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

anttijk commented on Source/WebCore/css/process-css-properties.py:
```diff
@@ -1125,41 +1130,34 @@ def aliases(self):
-    def __init__(self, kind, descriptor_method_name_modifier, descriptors):
-        self.kind = kind
-        self.descriptor_method_name_modifierdescriptor_method_name_modifier = descriptor_method_name_modifier
+class DescriptorsSet:
```
Is it really a DescriptorsSet and not a DescriptorSet?

weinig replied:
You are right. `DescriptorSet` would be a better name. Will update.

Commit:
Generalize descriptor processing to remove hardcoding in process-css-properties.py
https://bugs.webkit.org/show_bug.cgi?id=249730
rdar://103605497

Reviewed by Antti Koivisto.

Rather than adding explicit support for each new rule type (e.g.
@font-face, @counter-styles, etc.) to process-css-properties.py,
this generalizes all descriptors and utilizes the fact that the
rule type is present in the json already. While it is not common
to add new rule types, there was also no reason to hard code them.

* Source/WebCore/css/process-css-properties.py:

Canonical link: https://commits.webkit.org/258345@main
